arangod/Agency/RemoveFollower.cpp

```diff
@@ -230,11 +230,8 @@ bool RemoveFollower::start(bool&) {
                  });
 
   size_t inSyncCount = 0;
-  size_t notGoodCount = 0;
   for (auto const& pair : overview) {
-    if (pair.second == -1) {
-      ++notGoodCount;
-    } else if (static_cast<size_t>(pair.second) >= shardsLikeMe.size()) {
+    if (static_cast<size_t>(pair.second) >= shardsLikeMe.size()) {
```

Comment:
```suggestion
    if (pair.second != -1 && static_cast<size_t>(pair.second) >= shardsLikeMe.size()) {
```

For -1 it will be always true, so here your changes incorrect.

Reply:
Another option is cast `size()`/both to int/int64